### ISSUE TITLE
Tests fail with older perl versions

### ISSUE THREAD

**Issue by eserte** [kivilahtio/libhetula-client-perl#3]
Actually there are at least two failure modes:
* your module requires some features only available in perl 5.20 or so (signatures), but there's no specified minimum perl version in Makefile.PL and/or META files, so (e.g.) smokers happily send fail reports for these older perls
* by using Modern::Perl without a year specification always the newest feature set will be used, which means the module fails with 5.24 and older, too (and next year it will probably automatically fail with 5.26). The Modern::Perl doc recommends to specify a year in the `use` statement, see https://metacpan.org/pod/Modern::Perl#Forward-Compatibility

**Comment by eserte** [comment]
Another failure mode:
* t/01-usage.t fails if an older Modern::Perl (< 1.2018) is installed:
```
Unknown date '2018' requested
BEGIN failed--compilation aborted at t/01-usage.t line 5.
t/01-usage.t .. 
Dubious, test returned 255 (wstat 65280, 0xff00)
No subtests run 
```

**Comment by kivilahtio** [comment]
Thank you for your effort!
I will take a pass at the module perl requirements.
https://metacpan.org/pod/Modern::Perl
The docuentation states:
> use Modern::Perl;
> This enables the strict and warnings pragmas, as well as all of the features available in Perl 5.10.

> by using Modern::Perl without a year specification always the newest feature set will be used

I presume you meant the smokers assume the latest Perl?

I set Modern::Perl '2015' and specific the minimum perl version to 5.020 in the Makefile.PL and META.json
Hope it resolves this issue.

**Comment by eserte** [comment]
Yeah, it looks like the problem is fixed with the latest release.
